From: Salil Mehta <salil.mehta@huawei.com>
Date: Wed, 30 Aug 2017 12:06:03 +0100
Subject: net: hns3: Fixes the wrong IS_ERR check on the returned phydev value
Patch-mainline: v4.14-rc1
Git-commit: 752b06946cc99534a7a4a11b7196e9a1e3d7e098
References: bsc#1104353 FATE#326415

This patch removes the wrong check being done for the phy device being
returned by the mdiobus_get_phy() function. This function never returns
the error pointers.

Fixes: 256727da7395 ("net: hns3: Add MDIO support to HNS3 Ethernet
Driver for hip08 SoC")
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Salil Mehta <salil.mehta@huawei.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/hisilicon/hns3/hns3pf/hclge_mdio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/hisilicon/hns3/hns3pf/hclge_mdio.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3pf/hclge_mdio.c
@@ -150,7 +150,7 @@ int hclge_mac_mdio_config(struct hclge_d
 	}
 
 	phydev = mdiobus_get_phy(mdio_bus, mac->phy_addr);
-	if (!phydev || IS_ERR(phydev)) {
+	if (!phydev) {
 		dev_err(mdio_bus->parent, "Failed to get phy device\n");
 		mdiobus_unregister(mdio_bus);
 		return -EIO;
